src/Unosquare.RaspberryIO.Peripherals/AccelerometerGY521.cs

```diff
@@ -0,0 +1,205 @@
+    /// </summary>
+    public sealed class AccelerometerGY521 : IDisposable
+    {
+        private const int PwrMgmt1 = 0x6b;
+        private const int GyroSensitivity = 0x1b;
+        private const int AcclSensitivity = 0x1c;
+
+        private readonly ushort[] AFSSel = new ushort[] { 0x4000, 0x2000, 0x1000, 0x0800 };
+        private readonly byte[] FSSel = new byte[] { 0x83, 0x42, 0x20, 0x10 };
+
+        private readonly Thread ReadWorker;
+        private readonly TimeSpan ReadTime = TimeSpan.FromSeconds(2);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccelerometerGY521"/> class.
+        /// </summary>
+        /// <param name="device"> i2C device. </param>
+        public AccelerometerGY521(II2CDevice device)
+        {
+            Device = device;
+
+            // Reset sensor sleep mode to 0.
+            Device.WriteAddressByte(PwrMgmt1, 0);
+            Device.WriteAddressByte(GyroSensitivity, FSSel[(int)FSSEL.FSR250]);
+            Device.WriteAddressWord(AcclSensitivity, AFSSel[(int)AFSSEL.FSR2G]);
+
+            ReadWorker = new Thread(Run);
+        }
+
+        /// <summary>
+        /// Occurs when [data available].
+        /// </summary>
+        public event EventHandler<AccelerometerGY521EventArgs> DataAvailable;
+
+        /// <summary>
+        /// Retrieves the I2C Device.
+        /// </summary>
+        public II2CDevice Device { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance is running.
+        /// </summary>
+        /// <value><c>true</c> if this instance is running; otherwise, <c>false</c>.</value>
+        public bool IsRunning { get; private set; }
+
+        /// <summary>
+        /// Sets the sleep mode.
+        /// </summary>
+        public void SetSleepMode()
+        {
+            Device.WriteAddressByte(PwrMgmt1, 1);
+        }
+
+        /// <summary>
+        /// Starts this instance.
+        /// </summary>
+        public void Start()
+        {
+            IsRunning = true;
+            ReadWorker.Start();
+        }
+
+        /// <summary>
+        /// Stops this instance.
+        /// </summary>
+        public void Dispose()
+        {
+            if (IsRunning == false)
+                return;
+
+            StopContinuousReads();
+        }
+
+        /// <summary>
+        /// Retrieve the data capted by the sensor.
+        /// </summary>
+        /// <returns> Data calculated by GY-521 accelerometer. </returns>
+        public AccelerometerGY521EventArgs RetrieveSensorData()
+        {
+            // Read registry for gyroscope and accelrometer on the three axis.
+            var gyro = new Point3d(ReadWord2C(0x43), ReadWord2C(0x45), ReadWord2C(0x47));
+            var accel = new Point3d(ReadWord2C(0x3b), ReadWord2C(0x3d), ReadWord2C(0x3f));
+
+            return new AccelerometerGY521EventArgs(gyro, accel, Device.ReadAddressByte(GyroSensitivity), ReadWord2C(AcclSensitivity));
```

Comment:
You need to map Gyro and Accel scale range to the actual value:

| FS_SEL | Full Scale Range |
| -- | -- |
| 0 | ± 250 °/s |
| 1 | ± 500 °/s |
| 2 | ± 1000 °/s |
| 3 | ± 2000 °/s |

| AFS_SEL | Full Scale Range | 
| -- | -- |
| 0 | ± 2g |
| 1 | ± 4g |
| 2 | ± 8g |
| 3 | ± 16g |